packages/snaps-utils/src/versions.ts

```diff
@@ -1,12 +1,130 @@
-import { Json } from '@metamask/utils';
+import { assertStruct, Json } from '@metamask/utils';
 import {
   gt as gtSemver,
   maxSatisfying as maxSatisfyingSemver,
   satisfies as satisfiesSemver,
-  validRange as validRangeSemver,
+  valid as validSemVerVersion,
+  validRange as validSemVerRange,
 } from 'semver';
+import { is, refine, string, Struct, validate } from 'superstruct';
+import Opaque from 'ts-opaque';
```

Comment:
Why do we need. `ts-opaque` if we only use it in a few places? Wouldn't this function the same, without the need for a library:

```ts
export type SemVerRange = string & {
  readonly type: unique symbol;
};
```

Reply:
Because I'd be copy pasting the same types anyways to allow easy creation of Opaque types later.

Reply:
```ts
type Opaque<Type, Key extends symbol> = Type & { type: Key };

const version = Symbol('foo');
type Version = Opaque<'foo', typeof version>;

const range = Symbol('foo');
type Range = Opaque<'foo', typeof range>;
```

I don't think we need a library for this.